src/lib/source-control/git.ts

```diff
@@ -17,6 +18,35 @@ const git: SourceControlProvider = {
     const {stdout} = await run(`git rev-parse --show-toplevel`);
     return stdout.trim();
   },
+  getCurrentCommitMessage: async () => {
+    const root = await git.root();
+
+    const commitEditMessagePath = `${root}/.git/COMMIT_EDITMSG`;
+
+    if (fs.existsSync(commitEditMessagePath)) {
+        const commitEditMessage = fs
+            .readFileSync(commitEditMessagePath)
+            .toString();
+
+        const scissors = commitEditMessage.indexOf(
+            '------------------------ >8 ------------------------',
+        );
+
+        const end = scissors > -1 ? scissors : commitEditMessage.length;
+        const editMessage = commitEditMessage.substring(0, end).replace(/^#.*$/gm, '').trim();
+
+        if (editMessage.length) {
+            return editMessage;
+        }
+    }
+
+    return null;
+  },
+  getCommits: async (from, to) => {
+    const range = from && to ? `${from}..${to}` : '';
+    const { stdout: commitsString } = await run(`git log ${range} --pretty=format:"%s"`);
+    return commitsString.split('\n');
```

Comment:
I don't think this is what we want. This will only get the subject, I think we should get the whole body so we can add validations on trailers and other things. Because the messages have new lines in them, the trick is to use zero bytes

```bash
git log --pretty=format:"%B%x00"
```

We can then split on `\0` to get each message in a array.

Reply:
I get you. I'll apply your recommendation :+1: